### PR TITLE
Feature/setting screen

### PR DESCRIPTION
**Pull request by WalterGoldenOwl**
Description Pull Request:

## Todo
- Apply UI and logic setting screen

## Task
Link of task: 

1. https://trello.com/c/WekQnB82
2. https://trello.com/c/NA9JeU5T
3. https://trello.com/c/UK0Q66VU
4. https://trello.com/c/UySJJBZ8

## Screenshot
Screenshot_1698027209
Screenshot_1698027213
Screenshot_1698027216
Screenshot_1698027220
Screenshot_1698027264

## Video

https://github.com/dustin-goldenowl/jobspot/assets/144195318/9f3f2d5c-dbd3-4e23-8500-d946521e8103

### CODE REVIEW

File: lib/src/presentations/setting/data/repositories/setting_repository_impl.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+@LazySingleton(as: SettingRepository)
+class SettingRepositoryImpl extends SettingRepository {
+  @override
+  Future<DataState<bool>> deleteAccount() async {
+    try {
+      await Future.wait([
+        deletePost(),
+        deleteComment(),
+        FirebaseFirestore.instance
+            .collection("users")
```
Comment (dustin-goldenowl) [inline lib/src/presentations/setting/data/repositories/setting_repository_impl.dart:18]: Move to XCollection.users

File: lib/src/presentations/setting/data/repositories/setting_repository_impl.dart [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      ]);
+      return DataSuccess(true);
+    } catch (e) {
+      return DataFailed(e.toString());
+    }
+  }
+
+  Future deletePost() async {
+    final collection = await FirebaseFirestore.instance
+        .collection("posts")
```
Comment (dustin-goldenowl) [inline lib/src/presentations/setting/data/repositories/setting_repository_impl.dart:32]: The same above

File: lib/src/presentations/setting/data/repositories/setting_repository_impl.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        .get();
+    await Future.wait(collection.docs
+        .map((e) =>
+            FirebaseFirestore.instance.collection("posts").doc(e.id).delete())
+        .toList());
+  }
+
+  Future deleteComment() async {
+    final collection = await FirebaseFirestore.instance
+        .collection("comments")
```
Comment (dustin-goldenowl) [inline lib/src/presentations/setting/data/repositories/setting_repository_impl.dart:43]: The same above

Comment (WalterGoldenOwl) [reply lib/src/presentations/setting/data/repositories/setting_repository_impl.dart:43]: I fixed it in new branch and new PR

File: lib/src/presentations/setting/data/repositories/setting_repository_impl.dart (hunk 2 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/setting/data/repositories/setting_repository_impl.dart:32]: I fixed it in new branch and new PR

File: lib/src/presentations/setting/data/repositories/setting_repository_impl.dart (hunk 1 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/setting/data/repositories/setting_repository_impl.dart:18]: I fixed it in new branch and new PR

**Pull request merged**